addons/account_edi/models/account_edi_document.py

```diff
@@ -222,3 +216,31 @@ def _process_documents_web_services(self, job_count=None, with_commit=True):
             else:
                 if with_commit and len(jobs) > 1:
                     self.env.cr.commit()
+
+    def _process_documents_no_web_services(self):
+        """ Post and cancel all the documents that don't need a web service.
+        """
+        jobs = self.filtered(lambda d: not d.edi_format_id._needs_web_services())._prepare_jobs()
+        for documents, doc_type in jobs:
+            self._process_job(documents, doc_type)
+
+    def _process_documents_web_services(self):
+        ''' Post and cancel all the documents that need a web service. '''
+        jobs = self.filtered(lambda d: d.edi_format_id._needs_web_services())._prepare_jobs()
+        self._process_jobs(jobs, with_commit=False)
```

Comment:
Why would `with_commit` be False outside of tests?

Reply:
This method is only call by the `Send Now` button that is only callable on a single record.

Reply:
Edit: the CRON is using the `_cron_process_documents_web_services` method

Reply:
Both via the button and via the cron, you want to commit or at least do something to prevent a rollback, otherwise data you cannot loose could be lost.

Reply:
Indeed, it can only be called on one `account.move` record which could include many `account.edi.document`.